src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,18 +3,42 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int CAPACITY = 10;
+    private int currentPosition;
+    private final K[] keys;
+    private final V[] values;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[CAPACITY];
+        values = (V[]) new Object[CAPACITY];
+        currentPosition = 0;
+    }
 
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < currentPosition; i++) {
+            if (key != null && key.equals(keys[i]) || key == keys[i]) {
```

Comment:
```suggestion
            if (key == keys[i] || key != null && key.equals(keys[i])) {
```
from smaller to bigger, looks better)